scaaml/capture/input_generators/attack_point_iterator_models.py

```diff
@@ -0,0 +1,117 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pydantic models for the attack point iterator."""
+from pydantic import BaseModel, model_validator
+from typing import Any, Dict, Iterator, Literal, List, TypeAlias, Union
+
+from scaaml.capture.input_generators.attack_point_iterator_exceptions import ListNotPrescribedLengthException
+from scaaml.capture.input_generators.input_generators import balanced_generator, unrestricted_generator
+
+AttackPointIteratorT = Iterator[Dict[str, Any]]
+
+
+class ConstantIteratorModel(BaseModel):
+    """
+    Attack point iterator pydantic model that iterates over a constant.
+    
+    Args:
+        operation (str): The operation of the iterator
+        represents what the iterator does and what 
+        arguments should be present. This is only used once to
+        double check if the operation is the correct one.
+
+        name (str): The name represents the key name of the value.
+
+        length (int): The prescribed length for each list in values.
+        If one of the lists isn't the same length as this variable
+        it will raise an ListNotPrescribedLengthException.
+
+        values (List[List[int]]): List of lists of ints that gets
+        iterated through.
+    """
+    operation: Literal['constants']
+    name: str
+    length: int
+    values: List[List[int]]
+
+    @model_validator(mode='after')
+    def check(self) -> 'ConstantIteratorModel':
+        for value in self.values:
+            if len(value) != self.length:
+                raise ListNotPrescribedLengthException(
+                    f'The prescribed length is {self.length} and \
+                    the length of {value} is {len(value)}.')
+        return self
+
+    def len(self) -> int:
+        """Return the number of iterated elements."""
+        return len(self.values)
+
+    def iter(self) -> AttackPointIteratorT:
+        return iter({self.name: value} for value in self.values)
+
+    def get_generated_keys(self) -> List[str]:
+        return [self.name]
+
+
+class GeneratedIteratorModel(BaseModel):
+    """
+    Attack point iterator pydantic model that iterates over
+    the balanced and unrestricted generator.
+
+    operation (str): The operation of the iterator
+        represents what the iterator does and what 
+        arguments should be present. This is only used once to
+        double check if the operation is the correct one.
+
+    name (str): The name represents the key name of the value.
+    
+    length (int): Length of each yielded array for the generator.
+
+    bunches (int): How many bunches to return (yields `bunches * elements`
+        random arrays) for the generator.
+    
+    elements (int): Each array contains numbers in `range(elements)`
+        for the generator.
+    """
+
+    operation: Literal['balanced_generator', 'unrestricted_generator']
```

Comment:
NIT throughout this file. Why are we switching to using ' instead of "?
```suggestion
    operation: Literal["balanced_generator", "unrestricted_generator"]
```